efel/pyfeatures/pyfeatures.py

```diff
@@ -352,3 +355,148 @@ def phaseslope_max() -> np.ndarray | None:
+    """Time constant for an ion channel deactivation trace.
+    Fits for stim_start to stim_end as A + B * exp(-t/tau)."""
+    from scipy.optimize import curve_fit, OptimizeWarning
+
+    stim_start = _get_cpp_data("stim_start")
+    stim_end = _get_cpp_data("stim_end")
+    voltage = get_cpp_feature("voltage")
+    time = get_cpp_feature("time")
+
+    if voltage is None or time is None:
+        return None
+
+    # isolate stimulus interval
+    interval_indices = np.where((time >= stim_start) & (time < stim_end))
+    time_interval = time[interval_indices]
+    voltage_interval = voltage[interval_indices]
+
+    # correct time so that it starts from 0
+    time_interval -= time_interval[0]
+
+    # fit
+    try:
+        popt, _ = curve_fit(
+            exp_fit,
+            time_interval,
+            voltage_interval,
+            p0=(
+                1., voltage_interval[-1], max(
+                    0, voltage_interval[0] - voltage_interval[-1]
+                )
+            ),
+            bounds=(((0, -np.inf, 0), np.inf)),  # positive tau, positive A1
+            nan_policy="omit",
+        )
+    except (ValueError, RuntimeError, OptimizeWarning):
+        return None
+
+    return np.array([abs(popt[0])])
+
+
+def inactivation_time_constant() -> np.ndarray | None:
+    """Time constant for an ion channel inactivation trace.
+    Fits for trace maximum to stim end interval as A + B * exp(-t/tau)."""
+    from scipy.optimize import curve_fit, OptimizeWarning
+
+    stim_start = _get_cpp_data("stim_start")
+    stim_end = _get_cpp_data("stim_end")
+    voltage = get_cpp_feature("voltage")
+    time = get_cpp_feature("time")
+    # used to remove end of trace to remove artifacts due to stimulus change
+    end_skip = _get_cpp_data("inactivation_tc_end_skip")
+
+    if voltage is None or time is None:
+        return None
+
+    # isolate stimulus interval
+    stim_start_idx = np.flatnonzero(time >= stim_start)[0]
+    stim_end_idx = np.flatnonzero(time >= stim_end)[0]
+    time_interval = time[stim_start_idx:stim_end_idx - end_skip]
+    voltage_interval = voltage[stim_start_idx:stim_end_idx - end_skip]
+
+    # keep trace going from voltage max to stim end
+    # remove end of trace to remove artifacts due to stimulus change
+    max_idx = np.argmax(voltage_interval)
+    time_interval = time_interval[max_idx:]
+    voltage_interval = voltage_interval[max_idx:]
+
+    # correct time so that it starts from 0
+    if time_interval.size < 1:
+        return None
+    time_interval -= time_interval[0]
+
+    # fit
+    try:
+        popt, _ = curve_fit(
+            exp_fit,
+            time_interval,
+            voltage_interval,
+            p0=(1., voltage_interval[-1], voltage_interval[0] - voltage_interval[-1]),
+            bounds=(((0, -np.inf, 0), np.inf)),  # positive tau, positive A1
+            nan_policy="omit",
+        )
+    except (ValueError, RuntimeError, OptimizeWarning):
```

Comment:
I believe `OptimizeWarning` is a warning, not an exception, so I think it is not necessary to include it